src/components/button.js

```diff
@@ -77,55 +77,37 @@
     const compProps = isIcon ? iconButtonProps : buttonProps;
     const BtnComp = isIcon ? IconButton : Button;
 
-    const Comp = props => {
-      const { loading, onClick = () => {} } = props;
-      return (
-        <BtnComp
-          {...compProps}
-          startIcon={
-            !isIcon &&
-            icon !== 'None' &&
-            iconPosition === 'start' &&
-            React.createElement(Icons[icon])
-          }
-          endIcon={
-            !isIcon &&
-            icon !== 'None' &&
-            iconPosition === 'end' &&
-            React.createElement(Icons[icon])
-          }
-          {...props}
-          onClick={onClick}
-        >
-          {isIcon &&
-            React.createElement(Icons[icon === 'None' ? 'Error' : icon], {
-              fontSize: size,
-            })}
-          {!isIcon && buttonContent}
-          {!isIcon && loading && (
-            <CircularProgress size={16} className={classes.loader} />
-          )}
-        </BtnComp>
-      );
-    };
-
-    let ButtonComponent = <Comp />;
+    let onClickFn = () => {};
+    const [actionCallback, { loading }] = useAction(actionId);
+    if (isAction) onClickFn = actionCallback;
 
-    if (isAction) {
-      ButtonComponent = (
-        <B.Action actionId={actionId}>
-          {(callAction, { loading }) => {
-            const onClickAction = event => {
-              event.preventDefault();
-              if (!isDev && !loading && linkType === 'action') {
```

Comment:
Is this check not missing now? So now doing multiple clicks in a row might give unexpected behaviour?

Reply:
IMO, it was unnecessary. isAction in itself is sufficient enough.